src/marqo/tensor_search/add_docs.py

```diff
@@ -88,17 +88,17 @@ def threaded_download_and_preprocess_content(allocated_docs: List[dict],
                 if field not in tensor_fields:
                     continue
                 if isinstance(doc[field], str) or force_download:
-                    modality = infer_modality(doc[field])
-                    if (modality == Modality.IMAGE and is_no_index) or (
-                        modality == Modality.IMAGE and is_unstructured_index) or (
-                        is_structured_index and media_field_types_mapping[field] == FieldType.ImagePointer):
+                    inferred_modality = infer_modality(doc[field])
+                    if (inferred_modality == Modality.IMAGE and is_no_index) or (
+                        inferred_modality == Modality.IMAGE and is_unstructured_index) or (
+                        is_structured_index and media_field_types_mapping[field] == FieldType.ImagePointer and inferred_modality not in [Modality.TEXT, Modality.AUDIO, Modality.VIDEO]):
```

Comment:
Why not in if image is what you're after?

Reply:
removed this as discussed, infer_modality won't be used for structured image pointers